src/eventgenerator/generator/evaluator.go

```diff
@@ -102,10 +113,9 @@ func (e *Evaluator) retrieveAppMetrics(trigger *model.Trigger) ([]*model.AppMetr
 }
 func (e *Evaluator) sendTriggerAlarm(trigger *model.Trigger) {
 	url := e.scalingEngineUrl
-	b := new(bytes.Buffer)
-	json.NewEncoder(b).Encode(trigger)
-	path := strings.Replace(PathScale, "{appid}", trigger.AppId, 1)
-	resp, respErr := e.httpClient.Post(url+path, "", b)
+	jsonBytes, _ := json.Marshal(trigger)
```

Comment:
@fraenkel  Added the checking, but seems it can not be tested. I did some search on google and found that when struct contains map[int]int or channel will cause a marshal error. Currently we can not new a Trigger instance that contains these types.